packages/financial-templates-lib/src/logger/Logger.ts

```diff
@@ -31,15 +31,14 @@ import winston from "winston";
 import type { Logger as _Logger, LogEntry } from "winston";
 import type * as Transport from "winston-transport";
 import { createTransports } from "./Transports";
+import { delay } from "../helpers/delay";
 
 // This async function can be called by a bot if the log message is generated right before the process terminates.
-// By calling `await waitForLogger(Logger)`, with the local Logger instance, the process will wait for all upstream
-// transports to clear. This enables slower transports like slack to still send their messages before the process yields.
-// Note: typescript infers the return type to be unknown. This is fine, as the return type should be void and unused.
-export async function waitForLogger(logger: _Logger): Promise<unknown> {
-  const loggerDone = new Promise((resolve) => logger.on("finish", resolve));
-  logger.end();
-  return await loggerDone;
+// This method will check if the AugmentedLogger's isFlushed is set to true. If not, it will block until such time
+// that it has been set to true. Note that each blocking transport should implement this isFlushed bool to prevent
+// the logger from closing before all logs have been propagated.
+export async function waitForLogger(logger: AugmentedLogger) {
```

Comment:
goodbye broken `waitForLogger` and hello fixed version that enables transports to block! we can add this to all the other transports now.